hagrid/reservations/export_pdf.py

```diff
@@ -0,0 +1,342 @@
+        border_padding=(7, 2, 20),
+        word_wrap='LTR',
+        border_radius=None)
+
+
+class Document:
+
+    page = 0
+    w = 0
+    h = 0
+    right_inset = 75
+    bytes_buffer = None
+    canvas: canvas = None
+    side_label: str = ""
+    watermark: str = ""
+    cursor_y = 0
+    cursor_x = 0
+    ready = False
+
+    def __init__(self, filename: str, title: str, author: str, subject: str, side_label=""):
+        self.page = 0
+        self.ready = False
+        self.watermark = ""
+        self.bytes_buffer = BytesIO()
+        self.canvas = canvas.Canvas(self.bytes_buffer, pagesize=A4, pageCompression=0)
+        self.canvas.setTitle(title)
+        self.w, self.h = A4
+        self.canvas.setAuthor(author)
+        self.canvas.setSubject(subject)
+        self.side_label = side_label
+        self.new_page()
+        self.ready = True
+
+    def bookmark_page(self, text: str):
+        self.canvas.addOutlineEntry(text, "p" + str(self.page))
+        self.canvas.bookmarkPage("p" + str(self.page))
+
+    def render_page_hint(self):
+        self.canvas.drawString(self.w - 75, self.h - 35, "Page " + str(self.page))
+
+    def apply_watermark(self):
+        self.canvas.setFont("Helvetica", 30)
+        self.canvas.rotate(45)
+        self.canvas.setFillColorRGB(225, 225, 225, alpha=0.5)
+        self.canvas.drawString(75, 75, self.watermark)
+        self.canvas.setFillColorRGB(255, 255, 255, alpha=1.0)
+        self.canvas.rotate(360 - 45)
+        self.canvas.setFont("Helvetica", 14)
+
+    def set_watermark(self, watermark):
+        self.watermark = str(watermark)
+        self.apply_watermark()
+
+    def get_text_width(self, text: str):
+        return pdfmetrics.stringWidth(text, "Helvetica", 14)
+
+    def new_page(self, bookmark=None):
+        if self.ready:
+            self.canvas.showPage()
+        self.right_inset = 45
+        self.page += 1
+        self.cursor_y = self.h - 45
+        self.cursor_x = 45
+        self.canvas.setFont("Helvetica", 9)
+        self.canvas.rotate(90)
+        self.canvas.drawString(35, -25, self.side_label)
+        self.canvas.rotate(270)
+        self.canvas.setFont("Helvetica", 14)
+        if bookmark is not None:
+            self.bookmark_page(bookmark)
+        self.apply_watermark()
+        page_number_text = "Page " + str(self.page)
+        self.canvas.drawString(self.w - self.right_inset - self.get_text_width(page_number_text), 50, page_number_text)
+
+    def wrap_up(self):
+        self.canvas.showPage()
+        self.canvas.save()
+        pdf_bytes = self.bytes_buffer.getvalue()
+        self.bytes_buffer.close()
+        self.canvas = None
+        self.bytes_buffer = None
+        return pdf_bytes
+
+
+def timestamp(filestr=True):
+    if filestr:
+        return datetime.datetime.fromtimestamp(time.time()).strftime('%d_%m_%Y__%H_%M_Uhr')
+    else:
+        return datetime.datetime.fromtimestamp(time.time()).strftime('%m/%d/%Y %H:%M:%S Uhr')
```

Comment:
Please use ISO/international date format "YYYY-MM-DD HH:mm:ss" (no German) and use `slugify` like here

https://github.com/felixrindt/hagrid/blob/99fbe424e9b0820cf4ee601d22242da61136725d/hagrid/reservations/views/administration.py#L63

to make a filename-ready-string.